### ISSUE TITLE
Rework MSYS installation

### ISSUE THREAD

**Issue by Irvise** [thelfer/tfel#74]
The MSYS installation instructions (https://thelfer.github.io/tfel/web/install-windows-msys.html) refer to the original MSYS. We have MSYS2 now (MSYS is gone) and the links to mingw no longer work. I would propose to link directly to https://www.msys2.org/

Regards,
Fer

**Comment by thelfer** [comment]
Thanks. I'll have a look and try to reinstall from scratch a new version, adapting the documentation step by step.

**Comment by thelfer** [comment]
@Irvise I updated the instructions. Let me know if it's ok for you.

**Comment by Irvise** [comment]
Hi, sorry for the late reply.

I will try to document it as best I can:

1. Tfel is already available: https://github.com/msys2/MINGW-packages/blob/master/mingw-w64-tfel/PKGBUILD So a simple instruction that says "Run `pacman -S mingw-w64-x86_64-tfel`" should suffice.

But I did not stop there

Building tfel:
The receipt above uses CMake, which is the recommended method. However, the CMake available in MSYS2 uses Ninja by default, so the `-GNinja` flag is needed. If one wants to follow the general instructions available in the documentation, then the CMake flag `-G "MSYS Makefiles"` has to be added, so that CMake generates Makefiles.
It should also be noted that `gfortran` and `gnuplot` should be installed.

Now, there is a bit of an issue. MSYS2 supports several "backends". By that I mean that MSYS2 has different "roots" each using a different toolchain. For CMake to work as expected, one has to follow the instructions here: https://www.msys2.org/docs/cmake/ As it points out, you cannot simply do a `pacman -S cmake`, it HAS to be the `mingw-w64-x86_64-cmake` version. Same for Ninja and Make if those are used.
I can confirm that running the (current) instructions does not work and that the instructions provided in the link above (https://www.msys2.org/docs/cmake/) do work.

Also, the instruction to install git is duplicated, git is already included in the first code line :)

Now, getting into the actual build.
I tried to build from git/master. It builds but the tests fail. Here is the note I took:
```
Error on make check at 42% on YoungModulusTest_3.mfront

[ 42%] Built target EvolutionTest
[ 42%] Built target PipeTest
[ 42%] mfront source 
C:/msys64/home/foleo/tfel/mfront/tests/properties/YoungModulusTest_3.mfront
[ 42%] mfront source 
C:/msys64/home/foleo/tfel/mfront/tests/properties/ErrnoHandlingCheck.mfront
make[3]: *** 
[mfront/tests/properties/CMakeFiles/MFrontMaterialProperties.dir/build.make:265: 
mfront/tests/properties/src/YoungModulusTest_3.cxx] Error 127
make[3]: *** Waiting for unfinished jobs....
make[3]: *** 
[mfront/tests/properties/CMakeFiles/MFrontMaterialProperties.dir/build.make:290: 
mfront/tests/properties/src/ErrnoHandlingCheck.cxx] Error 127
make[2]: *** [CMakeFiles/Makefile2:3044: 
mfront/tests/properties/CMakeFiles/MFrontMaterialProperties.dir/all] Error 2
make[2]: *** Waiting for unfinished jobs....
```
Running make with the `-i` flag (ignore errors) I got a tremendous amount of errors. Most test failed with: `.Exit code 0xc0000374 ` and segfaults among others.

What about the 4.0 release?
Builds just fine. Fails the compilation of tests:
```
[ 41%] Linking CXX executable mfront-SupportedTypesTest.exe

C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: 
CMakeFiles/mfront-SupportedTypesTest.dir/objects.a(SupportedTypesTest.obj):SupportedTypesTest.cxx:(.text$_ZN18SupportedTypesTest5test3Ev[_ZN18SupportedTypesTest5test3Ev]+0x27): 
undefined reference to 
`mfront::SupportedTypes::TypeParsingOptions::TypeParsingOptions()'
```
Running `make -i` I got:
```
99% tests passed, 11 tests failed out of 4985

Total Test time (real) = 193.99 sec

The following tests FAILED:
         242 - 
brickGursonTvergaardNeedleman1982-test1_GursonTvergaardNeedleman1982_Tridimensional_UpWard_generic_mtest 
(Failed)
         243 - 
brickGursonTvergaardNeedleman1982-test1_GursonTvergaardNeedleman1982_Tridimensional_DownWard_generic_mtest 
(Failed)
         244 - 
brickGursonTvergaardNeedleman1982-test1_GursonTvergaardNeedleman1982_Tridimensional_ToNearest_generic_mtest 
(Failed)
         245 - 
brickGursonTvergaardNeedleman1982-test1_GursonTvergaardNeedleman1982_Tridimensional_TowardZero_generic_mtest 
(Failed)
         246 - 
brickGursonTvergaardNeedleman1982-test2_GursonTvergaardNeedleman1982_Tridimensional_UpWard_generic_mtest 
(Failed)
         247 - 
brickGursonTvergaardNeedleman1982-test2_GursonTvergaardNeedleman1982_Tridimensional_DownWard_generic_mtest 
(Failed)
         248 - 
brickGursonTvergaardNeedleman1982-test2_GursonTvergaardNeedleman1982_Tridimensional_ToNearest_generic_mtest 
(Failed)
         249 - 
brickGursonTvergaardNeedleman1982-test2_GursonTvergaardNeedleman1982_Tridimensional_TowardZero_generic_mtest 
(Failed)
         4755 - mfront-SupportedTypesTest (Not Run)
         4767 - mfront-query-test-0 (Failed)
         4768 - mfront-query-test-1 (Failed)
Errors while running CTest
Output from these tests are in: 
C:/msys64/home/foleo/tfel-build-4/Testing/Temporary/LastTest.log
Use "--rerun-failed --output-on-failure" to re-run the failed cases 
verbosely.
[100%] Built target check
```


Summary:
Tfel is already available in the official repos.
Tfel from git/master fails to run tests in a very bad way; but it does build.
Tfel release builds and runs most tests.
The instructions should be changed either to install the package from the repository or include the modifications to install the correct CMake, Ninja/Make.

If you need help, I could provide a patch, but do not expect it soon.

Regards,
Fer

**Comment by thelfer** [comment]
Hi @Irvise. Thanks for the feed-backs ! I'll make some seperate comments, if you don't mind.

**Comment by thelfer** [comment]
> 1. Tfel is already available: https://github.com/msys2/MINGW-packages/blob/master/mingw-w64-tfel/PKGBUILD So a simple instruction that says "Run `pacman -S mingw-w64-x86_64-tfel`" should suffice.

Yes, I knew it. The `install` instructions are meant for people who want to install the `master` branch or a specific version. But you're right, I do need to highlight the "official" packages !

**Comment by thelfer** [comment]
> I can confirm that running the (current) instructions does not work and that the instructions provided in the link above (https://www.msys2.org/docs/cmake/) do work.

That' evil. I did follow those instructions on a fresh MSYS2 installation on a fresh Windows installation. All went smoothly.

**Comment by thelfer** [comment]
> Also, the instruction to install git is duplicated, git is already included in the first code line :)

Ok. This is fixed.

**Comment by thelfer** [comment]
> I tried to build from git/master. It builds but the tests fail. Here is the note I took:

Yes, windows does not have RPATH, so we must update the PATH environment to find the generated libraries. This is done by the `env.sh` file which is automatically generated.

Would you try:

```bash
$ source env.sh
$ make check
```

**Comment by thelfer** [comment]
> What about the 4.0 release?
> Builds just fine. Fails the compilation of tests:

Yes. This is du to Issue #77. You may want to give `rliv-4.0` a try (It is fairly stable and only contains bugfixes).

**Comment by thelfer** [comment]
> The instructions should be changed either to install the package from the repository or include the modifications to install the correct CMake, Ninja/Make.

I would really appreciate your feedbacks on this and have some "always working" instructions, as the current ones did work for me.

**Comment by Irvise** [comment]
Hi @thelfer, thank you for your feedback.

Mmmmm... Weird that it worked for you directly... MMmmmmmm... I actually made sure to be running the MSYS2 shell to check those instructions and it did not work. I switched back to the MSYS2-MinGW one and following the instructions from the MSYS2 page (since I was surprised it was not working by default) and it worked.

Though, I must be honest, I tend to work in MinGW. Which means I install packages there and they may get mixed for one reason or another. So maybe the problem is on my side. It is rare now days that CMake fails like it did to me.

I will check master with the env.sh file once again. And I will try to provide as much feedback as necessary. If writing in this issue is longer than creating a patch, I will submit a PR directly.

Regards,
Fer

**Comment by thelfer** [comment]
@Irvise are we done ?

**Comment by Irvise** [comment]
Hi @thelfer,

I have not checked the comments you made. I have also not forgotten about this :)

I hope to have more time next week...

Regards,
Fer

**Comment by Irvise** [comment]
Hi,

I just wanted to say that I have still not had time to look into this. Luckily, today I got a new hard drive which gives me an opportunity to reinstall MSYS2 and start from a fresh install. I will not have time this week and most likely next week too, but I have not forgotten about it.

Regards,
Fer

**Comment by Irvise** [comment]
Hello! i come back with new results. This is going to be a long one. I logged pretty much everything that I did so as to leave no question unanswered. So, here it goes:

I installed a brand new MSYS2 system today. And then I updated it.

IMPORTANT NOTE: from now on, the only MSYS terminal that I used was the main MSYS2 terminal. I did NOT use the mingw, clang or any other terminal.

So, what did I do?

First: installed `git cmake base-devel gcc gcc-fortran`. I did NOT install gnuplot, python, boost, LaTeX nor doxygen.

Second: I tried compiling tfel. I used the master branch. For that I cloned it and ran `cmake ../tfel/ -Denable-fortran=ON -DCMAKE_INSTALL_PREFIX=/home/foleo/bin` on a clean build directory.

If you want the output, I can post it but I will not do so in order to not pollute this message.

After CMake ran, I proceeded with a `make -j3`. It failed. I have the complete output if you want it. The relevant error is:
```
In file included from /home/foleo/tfel-master/tfel/src/System/getFunction-win.cxx:14:
/home/foleo/tfel-master/tfel/src/System/getFunction.impl: In function ‘int (* tfel_getGenericBehaviourInitializeFunction(LibraryHandlerPtr, const char*))(mfront_gb_BehaviourData*, const mfront_gb_real*)’:
/home/foleo/tfel-master/tfel/src/System/getFunction.impl:196:10: error: invalid conversion from ‘int (*)(const mfront_gb_real*, mfront_gb_BehaviourData*)’ {aka ‘int (*)(const double*, mfront_gb_BehaviourData*)’} to ‘int (*)(mfront_gb_BehaviourData*, const mfront_gb_real*)’ {aka ‘int (*)(mfront_gb_BehaviourData*, const double*)’} [-fpermissive]
  196 |   return (int(TFEL_ADDCALL_PTR)(const mfront_gb_real *const,
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |          |
      |          int (*)(const mfront_gb_real*, mfront_gb_BehaviourData*) {aka int (*)(const double*, mfront_gb_BehaviourData*)}
  197 |                                 struct mfront_gb_BehaviourData *const))dlsym(l,
      |                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  198 |                                                                              f);
      |                                                                              ~~
/home/foleo/tfel-master/tfel/src/System/getFunction.impl: In function ‘int tfel_getUMATRequiresStiffnessTensor(LibraryHandlerPtr, const char*)’:
/home/foleo/tfel-master/tfel/src/System/getFunction.impl:35:10: warning: ‘char* strncat(char*, const char*, size_t)’ specified bound 24 equals source length [-Wstringop-overflow=]
   35 |   strncat(myname, "_requiresStiffnessTensor", 24u);
      |   ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/foleo/tfel-master/tfel/src/System/getFunction.impl: In function ‘int tfel_getUMATRequiresThermalExpansionCoefficientTensor(LibraryHandlerPtr, const char*)’:
/home/foleo/tfel-master/tfel/src/System/getFunction.impl:53:10: warning: ‘char* strncat(char*, const char*, size_t)’ specified bound 42 equals source length [-Wstringop-overflow=]
   53 |   strncat(myname, "_requiresThermalExpansionCoefficientTensor", 42u);
      |   ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [src/System/CMakeFiles/TFELSystem.dir/build.make:76: src/System/CMakeFiles/TFELSystem.dir/getFunction-win.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:2631: src/System/CMakeFiles/TFELSystem.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

It seems there is an error in the code. But more on that later.

Third: since the master branch was failing to compile, I decided to give the 4.0 release a go. I downloaded it from Github, unpacked and ran `cmake ../tfel-TFEL-4.0/ -Denable-fortran=ON -DCMAKE_INSTALL_PREFIX=/home/foleo/bin`. Once again, I have all the output. Then I ran `make -j3`. Which completed the compilation.

I then ran a `make check -j3` to run the tests. It failed to compile/link a library. Then I ran `source env.sh` and another `make check`. Which failed with:
```
[ 74%] Linking CXX executable mfront-SupportedTypesTest.exe
/usr/lib/gcc/x86_64-pc-msys/11.2.0/../../../../x86_64-pc-msys/bin/ld: CMakeFiles/mfront-SupportedTypesTest.dir/SupportedTypesTest.o:SupedTypesTest.cxx:(.text$_ZN18SupportedTypesTest5test3Ev[_ZN18SupportedTypesTest5test3Ev]+0x27): undefined reference to `mfront::Supportees::TypeParsingOptions::TypeParsingOptions()'
/usr/lib/gcc/x86_64-pc-msys/11.2.0/../../../../x86_64-pc-msys/bin/ld: CMakeFiles/mfront-SupportedTypesTest.dir/SupportedTypesTest.o:SupedTypesTest.cxx:(.text$_ZN18SupportedTypesTest5test3Ev[_ZN18SupportedTypesTest5test3Ev]+0x6c6): undefined reference to `mfront::Supportpes::TypeParsingOptions::~TypeParsingOptions()'
/usr/lib/gcc/x86_64-pc-msys/11.2.0/../../../../x86_64-pc-msys/bin/ld: CMakeFiles/mfront-SupportedTypesTest.dir/SupportedTypesTest.o:SupedTypesTest.cxx:(.text$_ZN18SupportedTypesTest5test3Ev[_ZN18SupportedTypesTest5test3Ev]+0xa41): undefined reference to `mfront::Supportpes::TypeParsingOptions::~TypeParsingOptions()'
/usr/lib/gcc/x86_64-pc-msys/11.2.0/../../../../x86_64-pc-msys/bin/ld: CMakeFiles/mfront-SupportedTypesTest.dir/SupportedTypesTest.o:SupedTypesTest.cxx:(.text$_ZN18SupportedTypesTest5test2Ev[_ZN18SupportedTypesTest5test2Ev]+0x1e): undefined reference to `mfront::Supportees::TypeParsingOptions::TypeParsingOptions()'
/usr/lib/gcc/x86_64-pc-msys/11.2.0/../../../../x86_64-pc-msys/bin/ld: CMakeFiles/mfront-SupportedTypesTest.dir/SupportedTypesTest.o:SupedTypesTest.cxx:(.text$_ZN18SupportedTypesTest5test2Ev[_ZN18SupportedTypesTest5test2Ev]+0x250): undefined reference to `mfront::Supportpes::TypeParsingOptions::~TypeParsingOptions()'
/usr/lib/gcc/x86_64-pc-msys/11.2.0/../../../../x86_64-pc-msys/bin/ld: CMakeFiles/mfront-SupportedTypesTest.dir/SupportedTypesTest.o:SupedTypesTest.cxx:(.text$_ZN18SupportedTypesTest5test2Ev[_ZN18SupportedTypesTest5test2Ev]+0x2b9): undefined reference to `mfront::Supportpes::TypeParsingOptions::~TypeParsingOptions()'
/usr/lib/gcc/x86_64-pc-msys/11.2.0/../../../../x86_64-pc-msys/bin/ld: CMakeFiles/mfront-SupportedTypesTest.dir/SupportedTypesTest.o:SupedTypesTest.cxx:(.text$_ZN18SupportedTypesTest5test2Ev[_ZN18SupportedTypesTest5test2Ev]+0x2d7): undefined reference to `mfront::Supportpes::TypeParsingOptions::~TypeParsingOptions()'
/usr/lib/gcc/x86_64-pc-msys/11.2.0/../../../../x86_64-pc-msys/bin/ld: CMakeFiles/mfront-SupportedTypesTest.dir/SupportedTypesTest.o:SupedTypesTest.cxx:(.text$_ZN18SupportedTypesTest5test1Ev[_ZN18SupportedTypesTest5test1Ev]+0x1f): undefined reference to `mfront::Supportees::TypeParsingOptions::TypeParsingOptions()'
/usr/lib/gcc/x86_64-pc-msys/11.2.0/../../../../x86_64-pc-msys/bin/ld: CMakeFiles/mfront-SupportedTypesTest.dir/SupportedTypesTest.o:SupedTypesTest.cxx:(.text$_ZN18SupportedTypesTest5test1Ev[_ZN18SupportedTypesTest5test1Ev]+0x625): undefined reference to `mfront::Supportpes::TypeParsingOptions::~TypeParsingOptions()'
/usr/lib/gcc/x86_64-pc-msys/11.2.0/../../../../x86_64-pc-msys/bin/ld: CMakeFiles/mfront-SupportedTypesTest.dir/SupportedTypesTest.o:SupedTypesTest.cxx:(.text$_ZN18SupportedTypesTest5test1Ev[_ZN18SupportedTypesTest5test1Ev]+0x689): undefined reference to `mfront::Supportpes::TypeParsingOptions::~TypeParsingOptions()'
/usr/lib/gcc/x86_64-pc-msys/11.2.0/../../../../x86_64-pc-msys/bin/ld: CMakeFiles/mfront-SupportedTypesTest.dir/SupportedTypesTest.o:SupedTypesTest.cxx:(.text$_ZN18SupportedTypesTest5test1Ev[_ZN18SupportedTypesTest5test1Ev]+0x6a7): undefined reference to `mfront::Supportpes::TypeParsingOptions::~TypeParsingOptions()'
collect2: error: ld returned 1 exit status
make[3]: *** [mfront/tests/unit-tests/CMakeFiles/mfront-SupportedTypesTest.dir/build.make:111: mfront/tests/unit-tests/mfront-SupportedsTest.exe] Error 1
make[2]: *** [CMakeFiles/Makefile2:3698: mfront/tests/unit-tests/CMakeFiles/mfront-SupportedTypesTest.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:2426: CMakeFiles/check.dir/rule] Error 2
make: *** [Makefile:199: check] Error 2
```

So I ran `make check -j3 -i` to ignore errors. Which finally ran all tests. The final results were:
```
[...]

99% tests passed, 1 tests failed out of 4989

Total Test time (real) = 481.69 sec

The following tests FAILED:
        4755 - mfront-SupportedTypesTest (Not Run)
Errors while running CTest
Output from these tests are in: /home/foleo/tfel-4.0/build/Testing/Temporary/LastTest.log
Use "--rerun-failed --output-on-failure" to re-run the failed cases verbosely.
[100%] Built target chec
```
The failed test had the following error:
```
         Start 4755: mfront-SupportedTypesTest
Could not find executable /home/foleo/tfel-4.0/build/mfront/tests/unit-tests/mfront-SupportedTypesTest.exe
Looked in the following places:
/home/foleo/tfel-4.0/build/mfront/tests/unit-tests/mfront-SupportedTypesTest.exe
/home/foleo/tfel-4.0/build/mfront/tests/unit-tests/mfront-SupportedTypesTest.exe.exe
/home/foleo/tfel-4.0/build/mfront/tests/unit-tests/Release/mfront-SupportedTypesTest.exe
/home/foleo/tfel-4.0/build/mfront/tests/unit-tests/Release/mfront-SupportedTypesTest.exe.exe
/home/foleo/tfel-4.0/build/mfront/tests/unit-tests/Debug/mfront-SupportedTypesTest.exe
/home/foleo/tfel-4.0/build/mfront/tests/unit-tests/Debug/mfront-SupportedTypesTest.exe.exe
/home/foleo/tfel-4.0/build/mfront/tests/unit-tests/MinSizeRel/mfront-SupportedTypesTest.exe
/home/foleo/tfel-4.0/build/mfront/tests/unit-tests/MinSizeRel/mfront-SupportedTypesTest.exe.exe
/home/foleo/tfel-4.0/build/mfront/tests/unit-tests/RelWithDebInfo/mfront-SupportedTypesTest.exe
/home/foleo/tfel-4.0/build/mfront/tests/unit-tests/RelWithDebInfo/mfront-SupportedTypesTest.exe.exe
/home/foleo/tfel-4.0/build/mfront/tests/unit-tests/Deployment/mfront-SupportedTypesTest.exe
/home/foleo/tfel-4.0/build/mfront/tests/unit-tests/Deployment/mfront-SupportedTypesTest.exe.exe
/home/foleo/tfel-4.0/build/mfront/tests/unit-tests/Development/mfront-SupportedTypesTest.exe
/home/foleo/tfel-4.0/build/mfront/tests/unit-tests/Development/mfront-SupportedTypesTest.exe.exe
home/foleo/tfel-4.0/build/mfront/tests/unit-tests/mfront-SupportedTypesTest.exe
home/foleo/tfel-4.0/build/mfront/tests/unit-tests/mfront-SupportedTypesTest.exe.exe
home/foleo/tfel-4.0/build/mfront/tests/unit-tests/Release/mfront-SupportedTypesTest.exe
home/foleo/tfel-4.0/build/mfront/tests/unit-tests/Release/mfront-SupportedTypesTest.exe.exe
home/foleo/tfel-4.0/build/mfront/tests/unit-tests/Debug/mfront-SupportedTypesTest.exe
home/foleo/tfel-4.0/build/mfront/tests/unit-tests/Debug/mfront-SupportedTypesTest.exe.exe
home/foleo/tfel-4.0/build/mfront/tests/unit-tests/MinSizeRel/mfront-SupportedTypesTest.exe
home/foleo/tfel-4.0/build/mfront/tests/unit-tests/MinSizeRel/mfront-SupportedTypesTest.exe.exe
home/foleo/tfel-4.0/build/mfront/tests/unit-tests/RelWithDebInfo/mfront-SupportedTypesTest.exe
home/foleo/tfel-4.0/build/mfront/tests/unit-tests/RelWithDebInfo/mfront-SupportedTypesTest.exe.exe
home/foleo/tfel-4.0/build/mfront/tests/unit-tests/Deployment/mfront-SupportedTypesTest.exe
home/foleo/tfel-4.0/build/mfront/tests/unit-tests/Deployment/mfront-SupportedTypesTest.exe.exe
home/foleo/tfel-4.0/build/mfront/tests/unit-tests/Development/mfront-SupportedTypesTest.exe
home/foleo/tfel-4.0/build/mfront/tests/unit-tests/Development/mfront-SupportedTypesTest.exe.exe
Unable to find executable: /home/foleo/tfel-4.0/build/mfront/tests/unit-tests/mfront-SupportedTypesTest.exe
```
The SupportedTypes issue is related to the failing `make check` compilation that I included above.

So I would say this was a success!!!

Fourth: I went back to the master branch. I ran a git blame on the file that was failing, see: https://github.com/thelfer/tfel/blame/2df255bec37f388d805ed554bef9accbdd279e66/src/System/getFunction.impl#L196
This indicates that the failing line was changed recently (Issues #24 #25). So I checked out the commit 921f67bfe2670896948bf92e0468cf63235bdba9. This commit is the one before the fixes from those issues.

I reran the cmake and make commands. It worked. I then sourced its env.sh file and ran the checks. All checks succeeded and there were no issues.

CONCLUSION

I think that the instructions in the MSYS2 page are correct and work. As it is said in the page, the build environment is minimal. I used gfortran too for example, and other tools could be added to get an even more complete installation.

It is also important to differentiate the "raw" MSYS2 system and the mingw provided one, which tends to be more complete and more widely used (that is where the official tfel package is offered).

So I give my two thumbs up to the changes that you did. Thank you!

P.S: the creep implementation is still on hold, and will stay there for the next month at the very least.

Regards,
Fer

**Comment by thelfer** [comment]
@Irvise I sincerly appreciate this kind of detailled feed-backs. I knew the issues with TFEL-4.0 and the trouble with SupportedTypes has been fixed in the `rliv-4.0` branch. Concerning the `master` branch, I'll fix that ASAP. Thank again !

P.S. For the creep implementation, let me know when you're ready, I'll be happy to help.
